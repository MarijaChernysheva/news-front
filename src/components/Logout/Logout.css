.logoutContent {
  display: flex;
  justify-content: right;
  align-items: center;
  padding-right: 15px;

}