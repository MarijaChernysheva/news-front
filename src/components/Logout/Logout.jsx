import React from 'react';
import { useDispatch } from 'react-redux';
import { useNavigate } from 'react-router';

import { grey } from '@mui/material/colors';
import Avatar from '@mui/material/Avatar';
import Stack from '@mui/material/Stack';
import PageviewIcon from '@mui/icons-material/Pageview';
import Button from '@mui/material/Button';

import { logoutClose } from '../../redux/actions';

import './Logout.css';

function Logout() {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const onClick = () => {
    dispatch(logoutClose());
    localStorage.removeItem('token');
    navigate('/');
  };

  const onClickAvatar = () => navigate('users/profile');

  return (
    <Stack className="logoutContent" direction="row" spacing={2}>
      <Avatar onClick={onClickAvatar} sx={{ bgcolor: grey[500] }}>
        <PageviewIcon />
      </Avatar>
      <Button variant="contained" onClick={onClick}>LOG OUT</Button>
    </Stack>
  );
}

export default Logout;
