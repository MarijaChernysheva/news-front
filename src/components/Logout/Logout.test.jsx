import '@testing-library/jest-dom';

import React from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render, screen, fireEvent } from '@testing-library/react';

import { Provider } from 'react-redux';
import { useNavigate } from 'react-router';
import { MemoryRouter } from 'react-router-dom';

import rootReducer from '../../redux/reducers/rootReducer';
import Logout from './Logout';

const store = configureStore({ reducer: rootReducer });
jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useNavigate: jest.fn(),
}));

const navigateMock = jest.fn();

const renderWithProviders = () => render(
  <MemoryRouter>
    <Provider store={store}>
      <Logout />
    </Provider>
  </MemoryRouter>,
);
describe('Logout', () => {
  beforeEach(() => {
    useNavigate.mockReturnValue(navigateMock);
  });
  test('should call localStorage.removeItem when the button is clicked', () => {
    renderWithProviders();
    const removeItemMock = jest.spyOn(Storage.prototype, 'removeItem');
    const btn = screen.getByRole('button');
    expect(btn).toBeInTheDocument();
    expect(btn).toMatchSnapshot();
    fireEvent.click(btn);
    expect(removeItemMock).toHaveBeenCalledWith('token');
    removeItemMock.mockRestore();
  });
  test('should call navigate when the avatar is clicked', () => {
    renderWithProviders();
    const avatar = screen.getByTestId('avatar');
    expect(avatar).toBeInTheDocument();
    fireEvent.click(avatar);
    expect(navigateMock).toHaveBeenCalledWith('users/profile');
  });
});
