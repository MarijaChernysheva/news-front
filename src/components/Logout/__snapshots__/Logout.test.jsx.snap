// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout should call localStorage.removeItem when the button is clicked 1`] = `
<button
  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  LOG OUT
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
